tests/visualization/test_pydot.py

```diff
@@ -0,0 +1,141 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import os
+import tempfile
+import unittest
+
+import retworkx
+from retworkx.visualization import pydot_draw
+
+try:
+    import pydot
+    import PIL
+
+    pydot.call_graphviz("dot", ["--version"], tempfile.gettempdir())
```

Comment:
Should this also be in the `HAS_PYDOT` check of ` retworkx/visualization/pydot.py`? (In this case, the would the current code raise at runtime from `pydot.graph_from_dot_data`?)

Reply:
I can add it over there, that might not be a bad idea. Right now it will error when we call `pydot.write()` because that's where graphviz gets called. But the error isn't actually helpful, so it's probably more user friendly to detect graphviz is missing and raise a meaningful error.

Reply:
:+1: if we can make the error message more meaningful. Realizing now though that this might have an impact on import performance, so worth weighing the best option..

Reply:
Well I didn't include the visualization submodule in the default import path on purpose because of this so it's not critical. But I was also thinking of just doing it at runtime (since it depends on having pydot present).

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/327/commits/f346cb1a7d1ffa7f3654645cac1ce957dd07358b